Assets/Scripts/Enemies/EnemyLaser/PlayerDetector.cs

```diff
@@ -0,0 +1,18 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class PlayerDetector : MonoBehaviour
+{
+    #region Private Functions
+
+    private void OnTriggerStay(Collider other)
+    {
+        if (other.tag == "MechaBody")
```

Comment:
Pro tip: use Tags.cs with a line like `if (other.tag == Tags.MechaBodyTag)` in case tags are changed